crates/ruff_linter/src/checkers/ast/analyze/definitions.rs

```diff
@@ -298,12 +304,30 @@ pub(crate) fn definitions(checker: &mut Checker) {
                 Rule::SectionUnderlineMatchesSectionLength,
                 Rule::SectionUnderlineNotOverIndented,
                 Rule::UndocumentedParam,
-            ]) {
-                pydocstyle::rules::sections(
-                    checker,
+            ]);
+            if enforce_sections || enforce_pydoclint {
+                let section_contexts = pydocstyle::helpers::get_section_contexts(
                     &docstring,
                     checker.settings.pydocstyle.convention.as_ref(),
                 );
+
+                if enforce_sections {
+                    pydocstyle::rules::sections(
+                        checker,
+                        &docstring,
+                        &section_contexts,
+                        checker.settings.pydocstyle.convention.as_ref(),
+                    );
+                }
+
+                if enforce_pydoclint {
+                    pydoclint::rules::check_docstring(
+                        checker,
+                        definition,
+                        &section_contexts,
+                        checker.settings.pydocstyle.convention.as_ref(),
```

Comment:
1. will the pydoclint rules get their own settings section, or will pydoclint re-use the existing [`pydocstyle.convention`](https://docs.astral.sh/ruff/faq/#does-ruff-support-numpy-or-google-style-docstrings) setting?
2. if DOC gets a `pydoclint` settings section, will the user have to duplicate `convention` directive?
3. will `convention` be a required config, or what will be the default?
4. will the pydoclint rules support `pep257` convention just like pydocstyle?

Reply:
I think we'll just reuse the `convention` setting from `pydocstyle` for now and document it as such in the `pydoclint` rules. Eventually, we'll want to recategorize all of these under a single `docstrings` section, so it'll be unified at some point in the future.

Reply:
4. Yes we will eventually add support for `pep257` and `sphinx` style docstrings